internal/services/communication/communication_service_email_domain_association_resource_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package communication_test
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/communicationservices"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/domains"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type CommunicationServiceEmailDomainAssociationResource struct{}
+
+func TestAccCommunicationServiceEmailDomainAssociationResource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_communication_service_email_domain_association", "test")
+	r := CommunicationServiceEmailDomainAssociationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCommunicationServiceEmailDomainAssociationResource_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_communication_service_email_domain_association", "test")
+	r := CommunicationServiceEmailDomainAssociationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_communication_service_email_domain_association"),
+		},
+	})
```

Comment:
we have a helper function for this test
```suggestion
		data.RequiresImportErrorStep(r.requiresImport),
	})
```